scripts/zones/QuBia_Arena/mobs/Doll_Factory.lua

```diff
@@ -1,28 +1,80 @@
 -----------------------------------
 -- Area: Qu'Bia Arena
---  Mob: Doll Factory
+-- Mob: Doll Factory
 -- BCNM: Factory Rejects
 -----------------------------------
 local entity = {}
 
-entity.onMobFight = function(mob, target)
-    local lastDoll = mob:getLocalVar("lastDoll")
-
-    if lastDoll == 0 or (lastDoll < 5 and GetMobByID(mob:getID() + lastDoll):isDead()) then
-        lastDoll = lastDoll + 1
-        SpawnMob(mob:getID() + lastDoll):updateEnmity(target)
-        mob:setLocalVar("lastDoll", lastDoll)
-        if lastDoll == 5 then
-            mob:setHP(0)
+local function immunity(mob, id)
+    if GetMobByID(id):isSpawned() then
+        mob:setMod(xi.mod.UDMGPHYS, -10000)
+        mob:setMod(xi.mod.UDMGMAGIC, -10000)
+        mob:setMod(xi.mod.UDMGRANGE, -10000)
+        mob:setMod(xi.mod.UDMGBREATH, -10000)
+    else
+        mob:setMod(xi.mod.UDMGPHYS, 0)
+        mob:setMod(xi.mod.UDMGMAGIC, 0)
+        mob:setMod(xi.mod.UDMGRANGE, 0)
+        mob:setMod(xi.mod.UDMGBREATH, 0)
+    end
+end
+
+local function spawnDoll(mob, id)
+    mob:setLocalVar("summoning", 1)
+    mob:entityAnimationPacket("casm")
+    mob:SetAutoAttackEnabled(false)
+    mob:SetMobAbilityEnabled(false)
+    mob:setMobMod(xi.mobMod.NO_MOVE, 1)
+    mob:timer(3000, function(mobArg)
+        if mob:isAlive() then
+            mobArg:entityAnimationPacket("shsm")
+            mobArg:SetAutoAttackEnabled(true)
+            mobArg:SetMobAbilityEnabled(true)
+            mob:setMobMod(xi.mobMod.NO_MOVE, 0)
+            SpawnMob(id):updateEnmity(mobArg:getTarget())
+            mob:setLocalVar("dollsLeft", mob:getLocalVar("dollsLeft") - 1)
+            mob:setLocalVar("dollAlive", 1)
+            mob:setLocalVar("summoning", 0)
         end
+    end)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setLocalVar("dollsLeft", 5)
+    mob:setLocalVar("currDoll", 1)
+    mob:setLocalVar("dollAlive", 0)
+end
+
+entity.onMobEngaged = function(mob)
+    mob:setLocalVar("timer", os.time() + math.random(12,20))
+end
+
+entity.onMobFight = function(mob, target)
+    local id = mob:getID() + mob:getLocalVar("currDoll")
+
+    if
+        mob:getLocalVar("timer") < os.time() and
+        mob:getLocalVar("dollsLeft") > 0 and
+        mob:getLocalVar("dollAlive") == 0 and
+        mob:getLocalVar("summoning") == 0
+    then
+        spawnDoll(mob, id)
+    end
+
+    if mob:getLocalVar("dollAlive") == 1 then
+        immunity(mob, id)
+    end
```

Comment:
Does this allow the player to deal damage to the detector while its in the middle of summoning a doll?

Reply:
Yes. only until the doll is alive will doll factory become invulnerable

Reply:
I would imagine the Doll Factory is completely immune to damage until all 5 dolls are defeated or else players could burst down the Factory in between doll spawns.

Reply:
I can change that, yeah! I just speculated it would be more difficult to not make him immune the entire fight because players lose the battlefield if he dies before he spawns all of his adds.

Reply:
https://ffxiclopedia.fandom.com/wiki/Factory_Rejects#:~:text=The%20Generic%20Doll%20will%20spawn%20one%20at%20a%20time%2C%20the%20Doll%20Factory%20must%20be%20kept%20alive%20until%20all%20Generic%20Dolls%20have%20been%20killed.

I think it may be a feature that he's killable throughout? However, as you and aerec saw in the capture, he is immune while they're alive

Reply:
@CriticalXI Do you want me to keep him as is? Or make him invincible until all dolls are out?

Reply:
When Aerec  tested this, the doll was immune at the start when it hasn't yet spawned any dolls so I believe he should be immune throughout the whole fight. I also believe the only lose condition would be the normal, wiping or timing out.

Reply:
okay! I'll make that fix now :)